resources/regexes.js

```diff
@@ -1,10 +1,202 @@
+      Regexes.maybeCapture(capture, 'z', f.z, '\\y{Float}') + ':' +
+      Regexes.maybeCapture(capture, 'heading', f.heading, '\\y{Float}') +
+      ':.*?:?$';
+    return Regexes.parse(str);
+  },
+
+  // fields: name, id, capture
+  headmarker: (f) => {
+    let capture = trueIfUndefined(f.capture);
+    let str = '\\y{TimeStamp} 1B:\\y{ObjectId}:' +
+      Regexes.maybeCapture(capture, 'name', f.name, '.*?') + ':....:....:' +
+      Regexes.maybeCapture(capture, 'id', f.id, '....') + ':';
+    return Regexes.parse(str);
+  },
+
+  // fields: name, capture
+  addedCombatant: (f) => {
+    let capture = trueIfUndefined(f.capture);
+    let str = '\\y{TimeStamp} 03:\\y{ObjectId}:Added new combatant ' +
+      Regexes.maybeCapture(capture, 'name', f.name, '.*?') + '\\.';
+    return Regexes.parse(str);
+  },
+
+  // fields: id, name, hp, x, y, z, capture
+  addedCombatantFull: (f) => {
+    let capture = trueIfUndefined(f.capture);
+    let str = '\\y{TimeStamp} 03:' + Regexes.maybeCapture(capture, 'id', f.id, '\\y{ObjectId}') +
+      ':Added new combatant ' + Regexes.maybeCapture(capture, 'name', f.name, '.*?') + '\\.' +
+      '.*?Max HP: ' +
+      Regexes.maybeCapture(capture, 'hp', f.hp, '[0-9]+') + '\.' +
+      '.*?Pos: \\(' +
+      Regexes.maybeCapture(capture, 'x', f.x, '\\y{Float}') + ',' +
+      Regexes.maybeCapture(capture, 'y', f.y, '\\y{Float}') + ',' +
+      Regexes.maybeCapture(capture, 'z', f.z, '\\y{Float}') + '\\)';
+    return Regexes.parse(str);
+  },
+
+  // fields: name, hp, capture
+  removingCombatant: (f) => {
+    let capture = trueIfUndefined(f.capture);
+    let str = '\\y{TimeStamp} 04:\\y{ObjectId}:Removing combatant ' +
+      Regexes.maybeCapture(capture, 'name', f.name, '.*?') + '\\.' +
+      '.*?Max HP: ' + Regexes.maybeCapture(capture, 'hp', f.hp, '[0-9]+') + '\.';
+    return Regexes.parse(str);
+  },
+
+  // fields: name, effect, source, duration, capture
+  gainsEffect: (f) => {
+    let capture = trueIfUndefined(f.capture);
+    let str = '\\y{TimeStamp} 1A:\\y{ObjectId}:' +
+      Regexes.maybeCapture(capture, 'name', f.name, '.*?') +
+      ' gains the effect of ' +
+      Regexes.maybeCapture(capture, 'effect', f.effect, '.*?') +
+      ' from ' +
+      Regexes.maybeCapture(capture, 'source', f.source, '.*?') +
+      ' for ' +
+      Regexes.maybeCapture(capture, 'duration', f.duration, '\\y{Float}') +
+      ' Seconds\\.';
+    return Regexes.parse(str);
+  },
+
+  // fields: name, effect, source, capture
+  losesEffect: (f) => {
+    let capture = trueIfUndefined(f.capture);
+    let str = '\\y{TimeStamp} 1E:\\y{ObjectId}:' +
+      Regexes.maybeCapture(capture, 'name', f.name, '.*?') +
+      ' loses the effect of ' +
+      Regexes.maybeCapture(capture, 'effect', f.effect, '.*?') +
+      ' from ' +
+      Regexes.maybeCapture(capture, 'source', f.source, '.*?') + '\\.';
+    return Regexes.parse(str);
+  },
+
+  // fields: name1, name2, id, capture
```

Comment:
Should these be source, target? I don't know how these log lines usually work.

Reply:
I wish!  My experience is that these are a bit arbitrary who is the source and who is the target of a tether.  Sometimes it's boss->player and sometimes player->boss.  That's why I went with `name1` and `name2` even though that's inconsistent with everything else.  If you had alternative name suggestions, I'd love to hear them, because I'm not happy with this either.

Reply:
That's frustrating; does it change between uses or just random quirks between bosses/encounters? It might still make sense to have them labeled as source and target, even if the wording is weird and can be denoted via comments?

I guess my fear here is that these will, regardless of name, end up victims of copy/paste and will still not make sense. For an example, imagine that common practice would follow that the player is always name2 (I don't know if this is the actual case), but then somewhere in coil it wasn't standardized that early (or something) and the boss is actually name2. It's still confusing, but harder (for me) to parse?

```
if (matches.source == data.me) {
   // WE ARE ACTUALLY SOURCE THIS TIME
  doThing();
}
```
vs
```
if (matches.name1 == data.me) {
   // WE ARE ACTUALLY NAME1 THIS TIME
  doThing();
}
```

But this is also a pretty strawman-y example so I'm not steadfast in it, just explaining my thought process.

Also, FWIW I'm just keeping consistent with source/target as I wrote that in the first comment, if there's better names then I'm all for it.

Reply:
It doesn't change between uses.  It's just random quirks between bosses and encounters.  I'd be fine calling them `source` and `target`.  Everything right now is just ordered the same way that the lines appear in the logs.

As a conversational aside, one thing that might help in the future is that I was planned to make some canned  conditions (and responses).  So, we could have like `Conditions.onMe()` but maybe there could be a tether version like `Conditions.tetherOnMe` or maybe somehow we could magically handle `onMe` internally and do both??

Reply:
Do you happen to know whether the source/target order seems to be tied to a specific tether type? I'll have to do some digging in my logs or possibly run some older stuff again, but just from a brief check through Grand Cosmos, both the "get away" and purple "okay" tethers are source --> target in the log lines, consistently. I'd be very interested to see whether the random tethers happen to be ones that are accompanied with overhead markers. (I'm thinking here of magnet tethers in particular, but this same thing is also true of the chain tethers in the Vault and Shinryu.)

Reply:
There's some entries in [the log guide](https://github.com/quisquous/cactbot/blob/master/docs/LogGuide.md#23-networktether) for examples, apparently (good job @quisquous for documenting!).

I _think_ (read: extrapolating off minimal data) that if you're tethered to something that isn't a tangible thing then  you are the source of the tether, which, to me, makes sense. For instance, based off this, I would assume the fire and ice tethers in Syrcus Tower's first boss would also have you as the source, etc.

Reply:
Hah, I even edited that very section a while ago and forgot the listed format was a thing.

Poking through my logs, now that you've suggested that, it looks like some inanimate objects might not actually *get* tether log lines? The Shivering/Smoldering/Shuddering Souls in Syrcus Tower certainly don't, instead placing a head marker on the tethered players. It's not clear where the status drops, I couldn't find corresponding status lines to go with the markers. (But that was a bit of a cursory look, it might need a thorough inspection.) The Chaosphere tethers in ARF similarly just... don't seem to do anything besides attacking with a bleed.

That being said, there were a couple places where it looks like your theory is correct. (Flight units on 9S, for instance.) This is all way beyond scope at this point, but I'll definitely look into that later.